src/btree/bt_sync.c

```diff
@@ -22,30 +22,32 @@ __sync_checkpoint_can_skip(WT_SESSION_IMPL *session, WT_REF *ref)
 
     WT_ASSERT_SPINLOCK_OWNED(session, &S2BT(session)->flush_lock);
 
-    mod = ref->page->modify;
-    txn = session->txn;
-
     /*
-     * We can skip some dirty pages during a checkpoint. The requirements:
+     * Don't skip checkpoint if at least one of the following conditions is met:
+     *
+     *  - This checkpoint happens during RTS, recovery or shutdown. Those scenarios should not leave
+     * anything dirty behind.
+     *  - This is the history store btree. As part of the checkpointing the data store, we will move
+     * the older values into the history store without using any transactions. This led to
+     * representation of all the modifications on the history store page with a transaction that is
+     * maximum than the checkpoint snapshot. But these modifications are done by the checkpoint
```

Comment:
Could you reword as history store checkpoint should not be dependent on the checkpoint snapshot.
Also, may be it is more readable, if the comment is split and moved to the corresponding checks.

Reply:
Addressed in efa117ecc3beed2ba7b2f3c0b3d7871b98375423 and 382d75c5c65ce6d853bd038987b6f77647efd0f7